src/main/java/org/mule/extension/http/api/request/builder/SimpleRequestHeader.java

```diff
@@ -32,7 +48,15 @@ public String getKey() {
 
   @Override
   public String getValue() {
-    return value;
+    return resolveLiteralExpression(value, expressionLanguage, watermark);
+  }
+
+  public void setExpressionLanguage(ExpressionLanguage expressionLanguage) {
+    this.expressionLanguage = expressionLanguage;
```

Comment:
let's avoid this kind of things. Injecting API classes such as this with core mule components is not something that would easily fit in new execution models such as voltron. Classes such as this should simply act as DTOs and business logic should be internal to the connector